src/main/java/sparksoniq/semantics/visitor/RuntimeIteratorVisitor.java

```diff
@@ -145,17 +145,17 @@
 import org.rumbledb.expressions.quantifiers.QuantifiedExpression;
 import org.rumbledb.expressions.quantifiers.QuantifiedExpressionVar;
 
-public class RuntimeIteratorVisitor extends AbstractExpressionOrClauseVisitor<RuntimeIterator> {
+public class RuntimeIteratorVisitor extends AbstractNodeVisitor<RuntimeIterator> {
 
     @Override
-    public RuntimeIterator visit(ExpressionOrClause expression, RuntimeIterator argument) {
+    public RuntimeIterator visit(Node expression, RuntimeIterator argument) {
```

Comment:
Should we also opt for 'node' variable name here as well or do you think expression is better fitting ?